### PR TITLE
docs: optimize unit testing

### PR DESCRIPTION
**Pull request by chaoranz758**
#### What type of PR is this?

docs

#### Check the PR title.

- [ ] This PR title match the format: \<type\>(optional scope): \<description\>
- [ ] The description of this PR title is user-oriented and clear enough for others to understand.

#### (Optional) More detailed description for this PR(en: English/zh: Chinese).

en: docs: optimize hertz unit testing
zh(optional):  docs: 优化 hertz 单元测试文档

### CODE REVIEW

File: content/zh/docs/hertz/tutorials/basic-feature/unit-test.md [hunk 1]
```diff
@@ -11,7 +11,74 @@ description: "Hertz 为用户提供的单元测试能力。"
...
-## 例子
+## 创建请求上下文
+
+```go
+func CreateUtRequestContext(method, url string, body *Body, headers ...Header) *app.RequestContext
+```
+
+### CreateUtRequestContext
+
+返回一个 app.RequestContext 对象，用于单元测试。
```
Comment (GuangmingLuo) [inline content/zh/docs/hertz/tutorials/basic-feature/unit-test.md:22]: app.RequestContext 缺少代码块 block

File: content/zh/docs/hertz/tutorials/basic-feature/unit-test.md [hunk 2]
```diff
@@ -44,9 +110,125 @@ func TestPerformRequest(t *testing.T) {
...
+	Flushed bool
+
+	result      *protocol.Response // cache of Result's return value
+	wroteHeader bool
+}
+```
+
+### NewRecorder
+
+返回初始化后的 ResponseRecorder 对象。
```
Comment (GuangmingLuo) [inline content/zh/docs/hertz/tutorials/basic-feature/unit-test.md:159]: 缺少代码块 block ，下同

Comment (chaoranz758) [reply content/zh/docs/hertz/tutorials/basic-feature/unit-test.md:159]: 我改一下

File: content/en/docs/hertz/tutorials/basic-feature/unit-test.md [hunk 3]
```diff
@@ -44,9 +110,125 @@ func TestPerformRequest(t *testing.T) {
 }
 ```
 
+## Receive Response
+
+When executing the [PerformRequest](#performarequest) function, functions such as [NewRecord](#newrecord), [Header](#header), [Write](#write), [WriteHeader](#writeheader), [Flush](#flush) have already been called internally. The user only needs to call the [Result](#result) function to obtain the returned `protocol.Response` object, and then perform unit testing.
+
+```go
+func NewRecorder() *ResponseRecorder
```
Comment (li-jin-gou) [inline content/en/docs/hertz/tutorials/basic-feature/unit-test.md:118]: 这块直接表格吧

**Review by li-jin-gou — APPROVED** [review]

**Pull request merged**